### PR TITLE
Pre-filter search results in selector

### LINKED ISSUE #29: Pre-filter search results in selector
The current implementation of search allows for a temporarily invalid list to be returned by the exported [`getSearchSelectors`](https://github.com/treasure-data/redux-search/tree/master/docs#getsearchselectorsresourcename-searchstateselector) in the event that a resource is removed from the searchable collection. The order of events is:

1. A search index is built and a search is run
2. A filtered results list is returned by the exported selector
3. One of the filtered resources is deleted (remove from the searchable collection)
4. This updates the state/store which in turn causes-
5. The search index is rebuilt and the filtered list is updated

In between steps 4 and 5 the state is temporarily invalid. The list returned by `getSearchSelectors` will still contain the id of the deleted resource. The data eventually corrects itself but the temporary state may cause problems.

After some discussion with Cesar we've agreed that the most reasonable solution to this would be for `getSearchSelectors` to automatically filter the list of ids before returning them- ensuring that all of the ids returned still exist in the searchable collection.

One caveat is that this may cause trouble in the event that a search index is created from derived data. To support this (uncommon?) use-case we'll also export a selector that does not automatically filter the returned list.

### PR DESCRIPTION
**Pull request by bvaughn**
Resolves #29

### CODE REVIEW

File: src/selectors.test.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+} from './actions'
+import reducer from './reducer'
+import {
+  getResultSelector,
+  getSearchSelectors,
+  getTextSelector,
+  getUnfilteredResultSelector
+} from './selectors'
+
+const RESOURCE_NAME = RESOURCE_NAME
```
Comment (cesarandreu) [inline src/selectors.test.js]: ???

Comment (cesarandreu) [inline src/selectors.test.js]: image

Comment (bvaughn) [inline src/selectors.test.js]: c

Comment (cesarandreu) [inline src/selectors.test.js]: I need to try that asap

**Pull request merged**